packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -434,9 +440,32 @@ export class StateMachine extends StateMachineBase {
     this.stateMachineArn = this.getResourceArnAttribute(resource.ref, {
       service: 'states',
       resource: 'stateMachine',
-      resourceName: this.physicalName,
+      resourceName: this.stateMachineName,
       arnFormat: ArnFormat.COLON_RESOURCE_NAME,
     });
+
+    if (graph.requiresExecutionPermissions) {
+      // this.grantStartExecution(this.role);
+      // this.grantExecution(this.role, 'states:DescribeExecution', 'states:StopExecution');
+      this.addToRolePolicy(new iam.PolicyStatement({
```

Comment:
I initially attempted to do this, but because of the circular dependency issue, I had to use wildcards and using `grantExecutionPermissions` doesn't allow using the wildcard instead of the state machine arn.